plugins/storage/volume/storpool/src/main/java/org/apache/cloudstack/storage/collector/StorPoolAbandonObjectsCollector.java

```diff
@@ -0,0 +1,320 @@
+                            PreparedStatement volumesOnHostpstmt = txn.prepareStatement("INSERT INTO `cloud`.`volumes_on_host1` (name, tag) VALUES (?, ?)");
+                            for (Map.Entry<String, String> volume : volumes.entrySet()) {
+                                if (volume.getValue().equals("volume")) {
+                                    addRecordToDb(volume.getKey(), pstmt, volume.getValue(), true);
+                                } else if (volume.getValue().equals("check-volume-is-on-host")) {
+                                    addRecordToDb(volume.getKey(), volumesOnHostpstmt, volume.getValue(), true);
+                                }
+                            }
+                            pstmt.executeBatch();
+                            volumesOnHostpstmt.executeBatch();
+                            String sql = "SELECT f.* FROM `cloud`.`volumes1` f LEFT JOIN `cloud`.`volumes` v ON f.name=v.path where v.path is NULL OR NOT state=?";
+                            findMissingRecordsInCS(txn, sql, "volume");
+
+                            String sqlVolumeOnHost = "SELECT f.* FROM `cloud`.`volumes_on_host1` f LEFT JOIN `cloud`.`storage_pool_details` v ON f.name=v.value where v.value is NULL";
+                            findMissingRecordsInCS(txn, sqlVolumeOnHost, "volumes_on_host");
+                        } catch (SQLException e) {
+                            log.info(String.format("[ignored] SQL failed due to: %s ",
+                                    e.getLocalizedMessage()));
+                        } catch (Throwable e) {
+                            log.info(String.format("[ignored] caught an error: %s",
+                                    e.getLocalizedMessage()));
+                        } finally {
+                            try {
+                                PreparedStatement pstmt = txn.prepareStatement("DROP TABLE `cloud`.`volumes1`");
+                                pstmt.executeUpdate();
+                                pstmt = txn.prepareStatement("DROP TABLE `cloud`.`volumes_on_host1`");
+                                pstmt.executeUpdate();
+                            } catch (SQLException e) {
+                                txn.close();
+                                log.info(String.format("createTemporaryVolumeTable %s", e.getMessage()));
+                            }
+                            txn.close();
+                        }
+                    }
+                });
+            }
+        }
+    }
+
+    class StorPoolSnapshotsTagsUpdate extends ManagedContextRunnable {
+
+        @Override
+        @DB
+        protected void runInContext() {
+            List<StoragePoolVO> spPools = storagePoolDao.findPoolsByProvider(StorpoolUtil.SP_PROVIDER_NAME);
+            Map<String, String> snapshots = new HashMap<String, String>();
+            if (spPools != null && spPools.size() > 0) {
```

Comment:
We could invert this condition (using `CollectionUtils#isEmpty`) and add a return to reduce the indentation of the block.